Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Controllers/MixedRealityControllerVisualizer.cs

```diff
@@ -0,0 +1,285 @@
+            }
+        }
+
+        #endregion SourcePose Event Handlers
+
+        #region Input Handlers
+
+        /// <summary>
+        /// Visualize the pressed button in the controller model, if supported
+        /// </summary>
+        /// <remarks>
+        /// Reserved for future implementation
+        /// </remarks>
+        /// <param name="eventData"></param>
+        public void OnInputDown(InputEventData eventData)
+        {
+            //Visualize button down
+        }
+
+
+        /// <summary>
+        /// Visualize the released button in the controller model, if supported
+        /// </summary>
+        /// <remarks>
+        /// Reserved for future implementation
+        /// </remarks>
+        /// <param name="eventData"></param>
+        public void OnInputUp(InputEventData eventData)
+        {
+            //Visualize button up
+        }
+
+
+        /// <summary>
+        /// Visualize the held trigger in the controller model, if supported
+        /// </summary>
+        /// <remarks>
+        /// Reserved for future implementation
+        /// </remarks>
+        /// <param name="eventData"></param>
+        public void OnInputPressed(InputEventData<float> eventData)
+        {
+            //Visualize single axis controls
+        }
+
+
+        /// <summary>
+        /// Visualize the movement of a dual axis input in the controller model, if supported
+        /// </summary>
+        /// <remarks>
+        /// Reserved for future implementation
+        /// </remarks>
+        /// <param name="eventData"></param>
+        public void OnPositionInputChanged(InputEventData<Vector2> eventData)
+        {
+            //Visualize dual axis controls
+        }
+
+        #endregion Input Handlers
+
+        #region Enable / Disable
+
+        /// <summary>
+        /// When the visualizer is enabled, create any attached controllers in the scene.
+        /// </summary>
+        /// <remarks>
+        /// Visualizers must start enabled in the scene in order to capture the controllers from the platform.
+        /// </remarks>
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+
+            if (MixedRealityManager.Instance.ActiveProfile.ControllersProfile != null && MixedRealityManager.Instance.ActiveProfile.ControllersProfile.RenderMotionControllers)
+            {
+                CreateControllerInstance(Handedness.Left);
+                CreateControllerInstance(Handedness.Right);
+            }
+        }
+
+        /// <summary>
+        /// If the visualizer is disabled, ensure all rendered controllers are cleaned up
+        /// </summary>
+        protected override void OnDisable()
+        {
+            DestroyControllerInstance(Handedness.Left);
+            DestroyControllerInstance(Handedness.Right);
+
+            base.OnDisable();
+        }
+
+        #endregion Enable / Disable
+
+        #region Controller Visualization Functions
```

Comment:
Functions -> Methods?

(A function usually denotes that there's always a return value)